ikarus/finiteelements/ferequirements.hh

```diff
@@ -187,7 +187,7 @@ namespace Ikarus {
   template <typename ParameterType = double>
   class ResultTypeMap {
   public:
-    using ResultArray = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, 0, 3, 3>;
+    using ResultArray = Eigen::Matrix<ParameterType, Eigen::Dynamic, Eigen::Dynamic, 0, 9, 3>;

```

Comment:
we can also decide here in the future to simply have here an `std::array<double,X>` and than working with the memory using https://eigen.tuxfamily.org/dox/group__TutorialMapClass.html. But I think this may lead to too much confusion. But anyway I think using a 9 there is good since it covers the case of unsymmetric matrices in Voigt notation. 

But anyway, the sizes we output here, should be the same as the ones we can actually write to VTK, since it does not really help to return 9 numbers here, but than indirectly store it in vtk somewhere.

Maybe it is even better to throw away this resulttype map think and then just give the result request as a template. Like

```cpp
fe.calculateAt<Result::LinearStress>(feRequirements,local);

```
Than the actual size is known at compile time. 
My idea was to not have again everythink templated, for better compile times and also it may be beneficial to request several results at once for performance but I'm not fully convinced anymore here.

This does not need to be fixed in this PR but the current situation is not so nice. What do you think?

Reply:
I thought about something like that too. For me it was a bit counter intuitive that the result is stored again with other results. Also we then have these two functions (out of my head)
```cpp 
getResult(Resulttype& resT);
getSingleResult();
```
which is a bit overcomplicated. 

The good thing with the approach `fe.calculateAt<Result::LinearStress>(feRequirements,local);` we can `constexpr` the parts of the function that correspond to the requessted quantity. 

But yeah i would do a secondary PR for that (after @tarun-mitruka 's PR is also merged :) )

Reply:
Yes, I think we should discuss this further as in `dune/grid/io/file/vtk/common.hh`, the size of the vector that can be written to VTK is restricted to 3, for example. We should also probably write more tests to see if it works correctly. Also, for 3D case, where the stress tensor is symmetric, in Voigt notation probably a vector of size 6 still cannot be used to write a VTK, or am I wrong?

Reply:
Dune-vtk assumes you have 9 components, if your provided vector is bigger than 3. https://gitlab.dune-project.org/search?search=tensor&nav_source=navbar&project_id=996&group_id=96&search_code=true&repository_ref=master

Reply:
> Dune-vtk assumes you have 9 components, if your provided vector is bigger than 3. https://gitlab.dune-project.org/search?search=tensor&nav_source=navbar&project_id=996&group_id=96&search_code=true&repository_ref=master

I think that should or could be handled by the `ResultFunction` or a specific `ResultEvaultor`

Reply:
I think the trailing zeros are added by dune-vtk, so if you provide 4 results, then in vtk you can directly view all 9 components whereas the last 5 are zeros